pkg/segment/aggregations/evalaggs.go

```diff
@@ -23,297 +23,543 @@ import (
+		}
+		currResult, exists := measureResults[measureAgg.String()]
+		result, err := PerformEvalAggForSum(measureAgg, countStat.Count, exists, currResult, fieldToValue)
+		if err != nil {
+			return fmt.Errorf("ComputeAggEvalForSum: Error while performing eval agg for sum, err: %v", err)
+		}
+		measureResults[measureAgg.String()] = result
+	} else {
+		sst, ok := sstMap[fields[0]]
+		if !ok {
+			return fmt.Errorf("ComputeAggEvalForSum: applyAggOpOnSegments sstMap was nil for aggCol %v", measureAgg.MeasureCol)
+		}
 
-	length := len(sst.Records)
-	for i := 0; i < length; i++ {
-		fieldToValue := make(map[string]utils.CValueEnclosure)
-		// Initialize fieldToValue
-		for _, field := range fields {
-			sst, ok := sstMap[field]
-			if !ok {
-				return fmt.Errorf("ComputeAggEvalForCount: applyAggOpOnSegments sstMap was nil for aggCol %v", measureAgg.MeasureCol)
+		length := len(sst.Records)
+		for i := 0; i < length; i++ {
+			fieldToValue = make(map[string]utils.CValueEnclosure)
+			err := PopulateFieldToValueFromSegStats(fields, measureAgg, sstMap, fieldToValue, i)
+			if err != nil {
+				return fmt.Errorf("ComputeAggEvalForSum: Error while populating fieldToValue from sstMap, err: %v", err)
 			}
 
-			if i >= len(sst.Records) {
-				return fmt.Errorf("ComputeAggEvalForCount: Incorrect length of field: %v for aggCol: %v", field, measureAgg.String())
+			currResult, exists := measureResults[measureAgg.String()]
+			result, err := PerformEvalAggForSum(measureAgg, uint64(length), exists, currResult, fieldToValue)
+			if err != nil {
+				return fmt.Errorf("ComputeAggEvalForSum: Error while performing eval agg for sum, err: %v", err)
 			}
-			fieldToValue[field] = *sst.Records[i]
+			measureResults[measureAgg.String()] = result
 		}
+	}
 
-		boolResult, err := measureAgg.ValueColRequest.BooleanExpr.Evaluate(fieldToValue)
-		if err != nil {
-			return fmt.Errorf("ComputeAggEvalForCount: there are some errors in the eval function that is inside the count function: %v", err)
-		}
+	return nil
+}
+
+func PerformEvalAggForCount(measureAgg *structs.MeasureAggregator, count uint64, exists bool, currResult utils.CValueEnclosure, fieldToValue map[string]utils.CValueEnclosure) (utils.CValueEnclosure, error) {
+	fields := measureAgg.ValueColRequest.GetFields()
+	finalResult := utils.CValueEnclosure{
+		Dtype: utils.SS_DT_FLOAT,
+		CVal:  float64(0),
+	}
+	finalValue := float64(0)
 
-		if boolResult {
-			countVal++
+	if len(fields) == 0 {
+		finalValue = float64(count)
+	} else {
+		if measureAgg.ValueColRequest.BooleanExpr != nil {
+			boolResult, err := measureAgg.ValueColRequest.BooleanExpr.Evaluate(fieldToValue)
+			if err != nil {
+				return currResult, fmt.Errorf("PerformEvalAggForCount: there are some errors in the eval function that is inside the count function: %v", err)
+			}
+			if boolResult {
+				finalValue++
+			}
+		} else {
+			// return count as 1 if expression is not boolean
+			finalValue = 1
 		}
```

Comment:
I don't understand this. Why is it not `finalValue++` on both 414 and 418?

Reply:
You can use either as this function would set the final value to 1 in two cases when a bool expression evaluates to true or just defaults to 1 if it is not a boolean expression. finalValue++ is equivalent to finalValue = 1 in this scenario.

Reply:
addressed in the latest commit.